### PR TITLE
Field disabled state follow form disabled state

### PR DESCRIPTION
**Pull request by lolcabanon**
Field will be disabled if the containing form is disabled.

### CODE REVIEW

**Comment by lolcabanon** [comment]
Ok put this one on hold there seems to be an issue...

I have an error `TypeError: Cannot read properties of undefined (reading 'value')` unless I disable an re-enable the form.

image

After I disable an re-enable the form it seems to work all good tho...

**Comment by lolcabanon** [comment]
Ok after a second checkup, it seems to work flawlessly as is with Budibase `2.5.2`.

Weird behaviour in my [last comment](#issuecomment-1511734994) only happen in version `2.4.31` (from that I cant test).

File: src/Component.svelte [hunk 1]
```diff
@@ -160,7 +160,7 @@
       {max}
       {placeholder}
       defaultDate={defaultValue}
-      {disabled}
```
Comment (melohagan) [inline src/Component.svelte:162]: I think this needs to be an or, i.e.

`disabled={fieldState?.disabled || disabled}`

**Review by melohagan — CHANGES_REQUESTED** [review]
Thanks for adding this - just needs a little tweak! Could you also increment the minor version in the `package.json` file so that the workflow kicks off please.

**Comment by lolcabanon** [comment]
Should it really be `||` ?

According to [MDN about `??`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) :

>The nullish coalescing (??) operator is a logical operator that returns its right-hand side operand when its left-hand side operand is [null](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/null) or [undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined), and otherwise returns its left-hand side operand.
>[...]
>The nullish coalescing operator can be seen as a special case of the [logical OR (||) operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR). The latter returns the right-hand side operand if the left operand is any [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) value, not only null or undefined. In other words, if you use || to provide some default value to another variable foo, you may encounter unexpected behaviors if you consider some falsy values as usable (e.g., '' or 0).

I think if `fieldState.disabled` is `false` it should return false indeed. But if either `fieldState` or `fieldState.disabled` are `undefined` it should then return `disabled` (local variable defaulting to `false`).

And by adding `?.` it is safe to test as `fieldState?.disabled` will return `undefined` if `fieldState` or `fieldState.disabled` are `null` or `undefined`.

**Review by melohagan — APPROVED** [review]
Tested it locally and works well! Thank you 😄

**Pull request merged**